discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -203,10 +226,255 @@ def process_create_userbank_instruction(
         logger.error(e)
 
 
+class PurchaseMetadataDict(TypedDict):
+    price: int
+    type: PurchaseType
+    id: int
+
+
+def get_purchase_metadata_from_memo(
+    session: Session, memos: List[str], slot: int
+) -> Union[PurchaseMetadataDict, None]:
+    """Checks the list of memos for one matching the format of a purchase's content_metadata, and then uses that content_metadata to find the premium_conditions associated with that content to get the price"""
+    for memo in memos:
+        logger.debug(f"index_user_bank.py | MEMO {memo}")
+        try:
+            content_metadata = json.loads(memo)
+            if "type" in content_metadata and "id" in content_metadata:
+                premium_conditions = None
+                if content_metadata["type"] == "TRACK":
+                    premium_conditions = (
+                        session.query(Track.premium_conditions)
+                        .filter(
+                            Track.track_id == content_metadata["id"],
+                            Track.is_current == True,
```

Comment:
Do we need checks for is_delete and is_unlisted here, or are we dong those elsewhere?

Reply:
Philosophical question: If the money is already sent for a purchase but the content was deleted/unlisted in the meantime, should the purchase succeed and you see a deleted/unlisted track in your purchased content, or should it fail (note that failing won't give you your money back)?

Reply:
Yeah, it's a good point... we can't return your money anyways, so we may as well let you 'purchase it' and then show it as disabled in the UI, versus just marking it as a transfer. Of course, this should never happen and client should always check to enforce track is still purchasable